tinyagentos/projects/canvas/watch_projection.py

```
@@ -0,0 +1,155 @@
+        value = payload.get(key)
+        if isinstance(value, str):
+            return value
+    return ""
+
+
+def _domain_of(url: Any) -> str:
+    if not isinstance(url, str) or not url.strip():
+        return ""
+    try:
+        return urlparse(url).netloc or ""
+    except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)